ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/coordinator/TransactionReaper.java

```diff
@@ -768,6 +757,30 @@ private final void shutdown(boolean waitForTransactions)
         _reaperWorkerThread = null;
     }
 
+    /**
+     * <p>
+     *     Wait for all the transactions to terminate normally.
+     * </p>
+     * <p>
+     *     Note:
+     *     this method assumes that the transaction system is not
+     *     currently capable of creating new transactions
+     *     (via TxControl.disable()), or we could be here for
+     *     a long time!
```

Comment:
I should have commented on this before but I would change "this method assumes" to "this method must only be used when" and I would remove "or we could be here for a long time!"
Why?
1. On the first part, it might be misleading because because there is another pitfall if it was capable (i.e. a transaction could be scheduled after `_reaperElements.isEmpty()` returns true)
2. On the second part, we can be a here a long time anyway (because the reaper timeouts are not shortening).

Reply:
Thanks Tom, done